tools/run_tests/run_tests.py

```diff
@@ -926,11 +926,31 @@ def test_specs(self):
                                  timeout_seconds=10 * 60,
                                  environ=_FORCE_ENVIRON_FOR_WRAPPERS)
         ]
-        tests.append(
-            self.config.job_spec(
-                ['tools/run_tests/helper_scripts/run_ruby_end2end_tests.sh'],
-                timeout_seconds=20 * 60,
-                environ=_FORCE_ENVIRON_FOR_WRAPPERS))
+        for test in [
+                'src/ruby/end2end/sig_handling_test.rb',
```

Comment:
is there a way to make sure that when new tests are added, we won't forget including them here?

Reply:
There isn't. But there wasn't a way to be sure this this previously when test cases were listed in the shell script, either.

One thing we could do which could perhaps clean this up and avoid listing test cases here is for `run_tests.py` to run every ruby script under `src/ruby/end2end` that ends with `_test.rb`.

Any thoughts?

Reply:
What I ended up doing in C# is to add an extra C# test ("c# sanity test") that only does one thing - lists all the existing C# tests and makes sure that matches https://github.com/grpc/grpc/blob/master/src/csharp/tests.json (which is where run_tests.py gets the list of C# tests from).
https://github.com/grpc/grpc/blob/fc4c9d74eebfdfbf99f181ebfd377f6a01db2946/src/csharp/Grpc.Core.Tests/SanityTest.cs#L44

For ruby you could do something similar - have a ruby tests.json, use it in run_tests.py and also check its contents by a dedicated ruby test (which reads the .json and also lists all the *_test.rb files and compares the two sets).